### PR TITLE
[WIP] add ACE support and TinyMCE4

### PR DESCRIPTION
**Pull request by zonky2**
add ACE support and TinyMCE4

### CODE REVIEW

**Comment by discordier** [comment]
The author validation shows trailing spaces, could you please correct this?

PS: please configure your IDE to trim trailing newlines on changes lines to prevent such issues in the future.

**Comment by xantippe** [comment]
fixed! thanks...

**Comment by stefanheimes** [comment]
Hmm looks good. But i didn't test it. Fix this the poblem that we have x script tags for the TinyMCE in the dom ?

**Comment by zonky2** [comment]
> Fix this the poblem that we have x script tags for the TinyMCE in the DOM?

No - the PR dont fix it; @discordier say, we should check every own Tiny configuration at every widget...

The primary goal was the implementation of ACE at MM - see https://github.com/MetaModels/core/issues/992

File: src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/ContaoWidgetManager.php [hunk 1]
```diff
@@ -158,7 +159,7 @@ public function preLoadRichTextEditor()
                 continue;
             }
 
-            if (strncmp($extra['rte'], 'tiny', 4) !== 0) {
+            if (strncmp($extra['rte'], 'tiny', 4) !== 0 && strncmp($extra['rte'], 'ace', 3) !== 0) {
                 continue;
             }
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/ContaoWidgetManager.php]: We should ensure here to only add each file once to solve the problem of adding the same config over and over again mentioned by @stefanheimes.

**Pull request merged**